language/move-lang/src/parser/mod.rs

```diff
@@ -40,29 +45,30 @@ pub(crate) fn parse_program(
     let mut source_definitions = Vec::new();
     let mut source_comments = CommentMap::new();
     let mut lib_definitions = Vec::new();
-    let mut errors: Errors = Errors::new();
+    let mut diags: Diagnostics = Diagnostics::new();
 
     for fname in targets {
-        let (defs, comments, es) = parse_file(&mut files, fname)?;
+        let (defs, comments, ds) = parse_file(&mut files, fname)?;
         source_definitions.extend(defs);
         source_comments.insert(fname, comments);
-        errors.extend(es);
+        diags.extend(ds);
     }
 
     for fname in deps {
-        let (defs, _, es) = parse_file(&mut files, fname)?;
+        let (defs, _, ds) = parse_file(&mut files, fname)?;
         lib_definitions.extend(defs);
-        errors.extend(es);
+        diags.extend(ds);
     }
 
-    let res = if errors.is_empty() {
+    // TODO fix this to allow warnings
```

Comment:
Is this a TODO for you before landing this, or for someone else to read later? What prevents warnings from appearing here?

Reply:
It's intentional. The parser pass is a bit funky and out of alignment with the code pattern of the other passes. It would take some refactoring to fix that to allow the non-blocking errors to go through, so that will be for a future PR